Fix typo in where rule options schema

DMartens opened the pull request:
The rule /where contains the typo "asssignment" in its rule options schema.
I would propose to add "additionalProperties: false" to each rule options schema, so users are warned if they mistype an option.

DMartens committed:
Fix typo in where rule options schema

The rule /where contains the typo "asssignment" in its rule options schema.
I would propose to add "additionalProperties: false" to each rule options schema, so users are warned if they mistype an option.

getify commented:
Hmmm... good catch. But now I'm wondering why this wasn't something caught by testing.

DMartens commented:
The rule options schema only states that the options are an object and the specified properties are boolean. It does not say anything about additional properties. In other words as the rule options schema is now it checks that "asssignment" (with the typo) is a boolean and "assignment" (without the typo)  can be anything as it is not specified. If you add the ` "additionalProperties": false` any property not present in the properties is considered an error and ESLint / its rule tester will throw an error.

getify commented:
Just so I make sure I understand: if I added that `additionalProperties: false` to the schema, but didn't fix the typo, would my tests have failed since the tests pass in correctly spelled `assignment`?

I'd like to make sure the underlying issue is fixed, so if you wouldn't mind amending your PR to include the additions to the various schemas, I'd happily accept the improvement/fix. Thanks! :)

DMartens committed:
Forbid additional rule option properties

Adding "additionalProperties: false" to each rule options schema so any additional/misspelled option is invalid.

DMartens commented:
Yes you understood correctly. I added the property to each rule options schema.
Glad I could be of help :)
